src/state/url-sync-mixin.ts

```diff
@@ -12,81 +17,76 @@ export const urlSyncMixin = <T extends Constructor<HassBaseEl>>(
   __DEMO__
     ? superClass
     : class extends superClass {
-        private _ignoreNextHassChange = false;
-
-        private _ignoreNextPopstate = false;
-
-        private _moreInfoOpenedFromPath?: string;
+        private _ignoreNextPopState = false;
 
         public connectedCallback(): void {
           super.connectedCallback();
           window.addEventListener("popstate", this._popstateChangeListener);
+          this.addEventListener("dialog-closed", this._dialogClosedListener);
         }
 
         public disconnectedCallback(): void {
           super.disconnectedCallback();
           window.removeEventListener("popstate", this._popstateChangeListener);
+          this.removeEventListener("dialog-closed", this._dialogClosedListener);
         }
 
-        protected hassChanged(newHass, oldHass): void {
-          super.hassChanged(newHass, oldHass);
-
-          if (this._ignoreNextHassChange) {
-            if (DEBUG) {
-              console.log("ignore hasschange");
-            }
-            this._ignoreNextHassChange = false;
-            return;
-          }
+        private _dialogClosedListener = (ev: HASSDomEvent<undefined>) => {
+          const dialogElement = ev.composedPath()[0] as Element;
```

Comment:
Would it be cleaner if we added this to the event details?

Reply:
I had that before, but is very error prone and a lot more work to implement it all 🤷‍♂️

Reply:
How can it be error prone? We fire it from the dialog element, so could use this.localName when passing it to `fireEvent` ?

Reply:
true...